portfolio/src/test/java/com/google/sps/BookResponseParserTest.java

```diff
@@ -0,0 +1,94 @@
+import java.lang.IllegalAccessException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+ 
+/**
+   This is a test class for RequestJson.java
+    Tests the implementations ability to convert
+    strings into a FullBook with the correct fields
+*/
+ 
+@RunWith(JUnit4.class)
+public final class BookResponseParserTest {
+  @Test
+  public void test1(){
+    //calls the API with the title and compares the result
+
+    String title = "A Court of Wings and Ruin";
+    ArrayList<String> authors = new ArrayList<String>();
+    authors.add("Sarah J. Maas");
+    String isbn = "9781619634497";
+    int pageCount = 432;
+    ArrayList<String> categories = new ArrayList<String>();
+    categories.add("Fiction");
+    String publisher = "Bloomsbury Publishing USA";
+    String publishedDate = "2017-05-02";
+    String maturityRating = "NOT_MATURE";
+    String thumbnail = "http://books.google.com/books/content?id=pLL-DAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api";
+    String language = "en";
+    String infoLink = "https://play.google.com/store/books/details?id=pLL-DAAAQBAJ&source=gbs_api";
+    String description = "Looming war threatens all Feyre holds dear in the third volume of the #1 "
+        +"New York Times bestselling A Court of Thorns and Roses series. Feyre has returned to the "
+        +"Spring Court, determined to gather information on Tamlin\'s maneuverings and the invading "
+        +"king threatening to bring Prythian to its knees. But to do so she must play a deadly game "
+        +"of deceit-and one slip may spell doom not only for Feyre, but for her world as well. As war "
+        +"bears down upon them all, Feyre must decide who to trust amongst the dazzling and lethal "
+        +"High Lords-and hunt for allies in unexpected places. In this thrilling third book in the "
+        +"#1 New York Times bestselling series from Sarah J. Maas, the earth will be painted red as "
+        +"mighty armies grapple for power over the one thing that could destroy them all.";
+
+    Book.Builder builder = Book.builder().title(title)
+            .categories(categories).authors(authors).language(language)
+            .description(description).infoLink(infoLink).pageCount(pageCount)
+            .publishedDate(publishedDate).publisher(publisher)
+            .maturityRating(maturityRating).thumbnail(thumbnail).isbn(isbn);
+    Book expected = builder.build();
+
+    String fileString = "";
+    try {
+      File file = new File("src/test/java/com/google/sps/BookParserTestFile.csv");
+      Scanner scanner = new Scanner(file, "utf-8");
+      while(scanner.hasNextLine()){
+          fileString += scanner.nextLine();
+      }
+    } catch (Exception ex) {
+      Assert.fail("Could not open file. " + ex);
+    }
```

Comment:
All the file resources must be closed after use to avoid resource leakage,  you could try-with-resources to auto-close the scanner. see: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html:

```
File file = new File("src/test/java/com/google/sps/BookParserTestFile.csv");
try (Scanner scanner = new Scanner(file, "utf-8")) {
    while (scanner. hasNextLine()) {
        fileString += scanner.nextLine();
    }
} 
```